tests/support/kubernetes.py

```diff
@@ -468,6 +522,127 @@ async def delete_namespaced_config_map(
         self._maybe_error("delete_namespaced_config_map", name, namespace)
         return self._delete_object(namespace, "ConfigMap", name)
 
+    # DEPLOYMENT API
+
+    async def create_namespaced_deployment(
+        self, namespace: str, body: V1Deployment
+    ) -> None:
+        """Create a deployment object.
+
+        This will create a ReplicaSet object, which will in turn create
+        a Pod object.
+
+        There is no parsing of ``replicas`` and no attempt to do proper
+        selectors; for purposes of the test, a replicaset and a single pod
+        with the same name as the deployment will be created.
+
+        Parameters
+        ----------
+        namespace
+            Namespace in which to create the deployment.
+        body
+            Deployment specification.
+
+        Raises
+        ------
+        ApiException
+            Raised with 409 status if the deployment already exists.
+        """
+        self._maybe_error("create_namespaced_pod", namespace, body)
+        name = body.metadata.name
+        await self.create_namespaced_replica_set(
+            namespace,
+            V1ReplicaSet(
+                metadata=V1ObjectMeta(name=name, namespace=namespace)
+            ),
+        )
+        body.status = V1DeploymentStatus(available_replicas=1)
+        self._update_metadata(body, "v1", "Deployment", namespace)
+        self._store_object(namespace, "Deployment", name, body)
```

Comment:
You should do these before creating the `ReplicaSet` because that's the order in which things happen in Kubernetes, and you also want to do the error checking done by `_update_metadata` and `_store_object` first.

Reply:
Deployment/Replicaset have gone away for now, because I don't need them.